### ISSUE TITLE
Debug why mobilenet performs worse when using batching

### ISSUE THREAD

**Issue by ejolly** [cosanlab/py-feat#151]
*copied from slack*

1. Model is originally trained with images of sizes 600 - 800 pixels, which means that our pretrained model works the best on images of that size: https://github.com/vitoralbiero/img2pose/issues/9
2. The differences in pose info detected for image with different scales is NOT a problem of Rescaling, but rather a problem for the model, especially about model parameter choices
3. The error is actually within tolerance. Misses like 2-4 degrees, out of 180

**Comment by TiankangXie** [comment]
The error result from batch normalization on multiple GPUs. 
The general issue is that the function nn.DataParallel gives different results for data in batches / single observations in the situation of > 1 GPU. See more at https://github.com/pytorch/pytorch/issues/24391

For our MobileNet model we are calling nn.DataParallel on the model (see https://github.com/cosanlab/py-feat/blob/8d49e53f6dc5a99bb784f30d5dd00dfca523fde1/feat/detector.py#L168)

The reason we are calling the DataParallel function is that the original weights are trained with parallel gpus and can **only be loaded with nn Parallel** (I know it's a weird requirement). I have fixed it by modifying the key names in the pretrained weights (credit to https://github.com/bearpaw/pytorch-classification/issues/27)

There is no obvious deviations right now. However, as pointed out, there are still minor deviations in the AU predictions, which resulted from numerical imprecisions. The numerical deviations require us to pay attention to numerical type conversions / enforcements
